torch/csrc/jit/ir/ir.h

```diff
@@ -264,8 +264,24 @@ struct Value {
   //          %4 = g(%3)
   //          %5 = inplace_(%3)
   //          %6 = h(%5, %5)
+  // XXX: does not check scoping legality, consider using
+  // replaceAllUsesDominatedByNodeWith
   TORCH_API void replaceAllUsesAfterNodeWith(const Node* node, Value* newValue);
 
+  // Replaces all uses of this value with 'newValue' that are dominated by
+  // 'node'. replaceAllUsesAfterNodeWith does not check domination, so in the
```

Comment:
The comment shows why using `replaceAllUsesAfterNodeWith` may be wrong. Perhaps it is more direct to just show how `replaceAllUsesDominatedByNodeWith` would transform the example (so that people don't have to understand what `repalceAllUsesAfterNodeWith` is).